### PR TITLE
docs: update is trusted

### LINKED ISSUE #28837: `isTrusted` property in `Event` isn't indicated whether this event was generated by a user action
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Event/isTrusted

### What specific section or headline is this issue about?

property description

### What information was incorrect, unhelpful, or incomplete?

MDN says
> The isTrusted read-only property of the [Event](https://developer.mozilla.org/en-US/docs/Web/API/Event) interface is a boolean value that is true when the event was generated by a user action, and false when the event was created or modified by a script or dispatched via [EventTarget.dispatchEvent()](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent).

But the focus event is filed with isTrusted = true even if the focus is changed by calling `focus()` on a element in script without a user action, e.g. page calls `focus()` to focus a element right after it's loaded.

What outlined in the [spec](https://dom.spec.whatwg.org/#dom-event-istrusted) is:
> [isTrusted](https://dom.spec.whatwg.org/#dom-event-istrusted) is a convenience that indicates whether an [event](https://dom.spec.whatwg.org/#concept-event) is [dispatched](https://dom.spec.whatwg.org/#concept-event-dispatch) by the user agent (as opposed to using [dispatchEvent()](https://dom.spec.whatwg.org/#dom-eventtarget-dispatchevent)). The sole legacy exception is [click()](https://html.spec.whatwg.org/multipage/interaction.html#dom-click), which causes the user agent to dispatch an [event](https://dom.spec.whatwg.org/#concept-event) whose [isTrusted](https://dom.spec.whatwg.org/#dom-event-istrusted) attribute is initialized to false.

### What did you expect to see?

Perhaps something in line with the spec.

### Do you have any supporting links, references, or citations?

https://dom.spec.whatwg.org/#dom-event-istrusted

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by a0m0rajab**
### Description

This PR adds extra details to the **isTrusted** property to align with the specs.

### Motivation

This follows the next issue; https://github.com/mdn/content/issues/28837

### Additional details

[dom.spec.whatwg.org/#dom-event-istrusted](https://dom.spec.whatwg.org/#dom-event-istrusted)

### Related issues and pull requests

Fixes https://github.com/mdn/content/issues/28837

### CODE REVIEW

File: files/en-us/web/api/event/istrusted/index.md [hunk 1]
```diff
@@ -10,9 +10,11 @@ browser-compat: api.Event.isTrusted
 
 The **`isTrusted`** read-only property of the
 {{domxref("Event")}} interface is a boolean value that is `true`
-when the event was generated by a user action, and `false` when the event was
-created or modified by a script or dispatched via
+when the event was generated by a user action or user agent,
+and `false` when the event was dispatched via
 {{domxref("EventTarget.dispatchEvent()")}}.
+The only exception is click, which intialise the **`isTrusted`**
+property to false in user agents.
```
Comment (skyclouds2001) [inline files/en-us/web/api/event/istrusted/index.md:17]:
```suggestion
The only exception is `click` event, which intialise the **`isTrusted`**
property to `false` in user agents.
```

File: files/en-us/web/api/event/istrusted/index.md [hunk 2]
```diff
@@ -10,9 +10,11 @@ browser-compat: api.Event.isTrusted
 
 The **`isTrusted`** read-only property of the
 {{domxref("Event")}} interface is a boolean value that is `true`
-when the event was generated by a user action, and `false` when the event was
-created or modified by a script or dispatched via
+when the event was generated by a user action or user agent,
+and `false` when the event was dispatched via
```
Comment (skyclouds2001) [inline files/en-us/web/api/event/istrusted/index.md:14]:
```suggestion
when the event was generated by user agent,
and `false` when the event was dispatched via
```

following up the spec, maybe there only mention *user agent* is enough?

Comment (a0m0rajab) [reply files/en-us/web/api/event/istrusted/index.md:14]: Sure, I will follow tha.t

**Comment by a0m0rajab** [comment]
Thanks for the review @skyclouds2001

**Review by Josh-Cena — APPROVED** [review]
Thanks

File: files/en-us/web/api/event/istrusted/index.md [hunk 3]
```diff
@@ -10,9 +10,11 @@ browser-compat: api.Event.isTrusted
 
 The **`isTrusted`** read-only property of the
 {{domxref("Event")}} interface is a boolean value that is `true`
-when the event was generated by a user action, and `false` when the event was
-created or modified by a script or dispatched via
+when the event was generated by the user agent (including using programmatic methods, such as {{domxref("HTMLElement.focus()")}}),
```
Comment (Josh-Cena) [inline files/en-us/web/api/event/istrusted/index.md:13]:
```suggestion
when the event was generated by the user agent (including via user actions and programmatic methods such as {{domxref("HTMLElement.focus()")}}),
```

File: files/en-us/web/api/event/istrusted/index.md [hunk 4]
```diff
@@ -10,9 +10,11 @@ browser-compat: api.Event.isTrusted
 
 The **`isTrusted`** read-only property of the
 {{domxref("Event")}} interface is a boolean value that is `true`
-when the event was generated by a user action, and `false` when the event was
-created or modified by a script or dispatched via
+when the event was generated by the user agent (including via user actions and programmatic methods such as {{domxref("HTMLElement.focus()")}}),
+and `false` when the event was dispatched via
 {{domxref("EventTarget.dispatchEvent()")}}.
+The only exception is `click` event, which intialise the **`isTrusted`**
```
Comment (Josh-Cena) [inline files/en-us/web/api/event/istrusted/index.md:16]:
```suggestion
The only exception is the `click` event, which initializes the `isTrusted`
```

**Pull request merged**